tests/engine/test_debit_note_intervals.py

```diff
@@ -211,7 +210,7 @@ def test_verify_debit_note_intervals(
             {"details___ref__demand__properties": {PROP_DEBIT_NOTE_INTERVAL_SEC: 100}},
             {},
             lambda: datetime.now() - timedelta(seconds=300),
-            3,
+            4,
```

Comment:
Well, I think these require either:

* an additional test case - so that we have:
    * a test that fulfills the described criteria ("n-th debit note received earlier than expected, based on the agreed-upon interval"), and for which the results is _no_ agreement termination
    * and another test that sends nth+1 debit note earlier than expected... 
* or we should re-write the explanations, taking into consideration the updated logic... 

or, _generally_, maybe we should have a look again at those test cases and see whether they still make sense - maybe we should fix more descriptions and scenarios

Reply:
Added a change to high-end test cases. We have test cases for _just in range_ and _just outside the range._ The range just moved by one with the change to the logic (`num_notes - 1`)